lambda/index.js

```diff
@@ -8,13 +8,25 @@ const Sharp = require('sharp');
 
 const BUCKET = process.env.BUCKET;
 const URL = process.env.URL;
+const ALLOWED_RESOLUTIONS = process.env.ALLOWED_RESOLUTIONS ? new Set(process.env.ALLOWED_RESOLUTIONS.split(/\s*,\s*/)) : new Set([]);
 
 exports.handler = function(event, context, callback) {
   const key = event.queryStringParameters.key;
-  const match = key.match(/(\d+)x(\d+)\/(.*)/);
-  const width = parseInt(match[1], 10);
-  const height = parseInt(match[2], 10);
-  const originalKey = match[3];
+  const match = key.match(/((\d+)x(\d+))\/(.*)/);
+
+  //Check if requested resolution is allowed
+  if(0 != ALLOWED_RESOLUTIONS.size && !ALLOWED_RESOLUTIONS.has(match[1]) ) {
+    callback(null, {
+      statusCode: '403',
+      headers: {},
+      body: '',
+    });
+    return;
+  }
+
+  const width = parseInt(match[2], 10);
```

Comment:
Oh, and if you could move these up. I'm fine doing these assignments even if we bail. I think it reads better. Basically:

```javascript
const key = event.queryStringParameters.key;
const match = key.match(/((\d+)x(\d+))\/(.*)/);
const dimensions = match[1];
const width = parseInt(match[2], 10);
const height = parseInt(match[3], 10);
const originalKey = match[4];
```

Reply:
Sure no problem